projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java

```diff
@@ -1233,6 +1234,9 @@ public void createJob(
                     .runAsGroup(runAsGroup)
                     .fsGroup(fsGroup));
 
+    if (StringUtils.isNotEmpty(registrySecret)) {
+      podSpecBuilder.addNewImagePullSecret().withName(registrySecret).endImagePullSecret();
+    }
```

Comment:
nit: would keep handling the serviceAccountName before registrySecret, matching the parameters order, for better readability.